swarm/network/stream/delivery.go

```diff
@@ -209,10 +209,10 @@ R:
 
 		// this should be has locally
 		chunk, err := d.db.Get(req.Key)
-		if !bytes.Equal(chunk.Key, req.Key) {
-			panic(fmt.Errorf("processReceivedChunks: chunk key %s != req key %s (peer %s)", chunk.Key.Hex(), req.Key.Hex(), req.peer.ID()))
-		}
 		if err == nil {
+			if !bytes.Equal(chunk.Key, req.Key) {
+				panic(fmt.Errorf("processReceivedChunks: chunk key %s != req key %s (peer %s)", chunk.Key.Hex(), req.Key.Hex(), req.peer.ID()))
+			}
```

Comment:
Minor thing, not a blocker: go convention is to first check the error case. I would just start with the
`if err != nil`, then basically everything in the loop can be in that if branch, and then if err is nil then check for the bytes.Equal, and we don't even need the `continue R` then.

Reply:
:+1: this check is not needed in fact, remnants of THE BUG hunt.

Reply:
@zelig so can i remove the check?